src/components/input/Select.tsx

```diff
@@ -311,31 +320,50 @@ function useListboxPositioning(
       buttonDistanceToBottom < listboxHeight &&
       buttonDistanceToTop > buttonDistanceToBottom;
 
-    if (asPopover) {
-      const { top: bodyTop } = document.body.getBoundingClientRect();
-      const absBodyTop = Math.abs(bodyTop);
-
-      return setListboxCSSProps({
-        minWidth: `${buttonWidth}px`,
-        top: shouldListboxDropUp
-          ? `calc(${absBodyTop + buttonDistanceToTop - listboxHeight}px - ${LISTBOX_TOGGLE_GAP})`
-          : `calc(${absBodyTop + buttonDistanceToTop + buttonHeight}px + ${LISTBOX_TOGGLE_GAP})`,
-        left:
-          right && listboxWidth > buttonWidth
-            ? `${buttonLeft - (listboxWidth - buttonWidth)}px`
-            : `${buttonLeft}px`,
-      });
+    if (!asPopover) {
+      // Set styles for non-popover mode
+      if (shouldListboxDropUp) {
+        return setListboxCSSProps({
+          bottom: '100%',
+          marginBottom: LISTBOX_TOGGLE_GAP,
+        });
+      }
+
+      return setListboxCSSProps({ top: '100%', marginTop: LISTBOX_TOGGLE_GAP });
     }
 
-    // Set styles for non-popover mode
-    if (shouldListboxDropUp) {
-      return setListboxCSSProps({
-        bottom: '100%',
-        marginBottom: LISTBOX_TOGGLE_GAP,
-      });
+    const { top: bodyTop, width: bodyWidth } =
+      document.body.getBoundingClientRect();
+    const absBodyTop = Math.abs(bodyTop);
+
+    // Calculate space available for the listbox in regular conditions, to make
+    // sure it never increases the body size, which could cause horizontal
+    // scrollbars to appear
+    const regularAvailableSpace =
+      (right ? buttonLeft + buttonWidth : bodyWidth - buttonLeft) -
```

Comment:
The meaning of the existing `right` function parameter is as clear as it could be. This ultimately comes from `right` in `BaseSelectProps`. That has some documentation but it could be improved. The summary reads "Align the listbox to the right." To the right of _what_?

Reply:
I'm not sure if I follow. Are you suggesting to update the comment somewhere?